packages/apollo-gateway/src/snapshotSerializers/queryPlanSerializer.ts

```diff
@@ -142,3 +155,12 @@ function printNodes(
 
   return result;
 }
+
+function isQueryPlanOperation(op: OperationDefinitionNode) {
```

Comment:
It's possible for a user to write a query that passes the requirements in this predicate, accidentally falling into the `true` case. Is there a way we can tag an entities operation a bit more clearly, like an additional key on the object? If no, we could find the `_entities` field in the document with a visitor and that would be sufficient since federation "owns" that field and users shouldn't be writing queries to it.